ui/ducks/send/send.js

```diff
@@ -2259,6 +2261,19 @@ export function getRecipientUserInput(state) {
 }
 
 export function getRecipient(state) {
+  const checksummedAddress = toChecksumHexAddress(
+    state[name].recipient.address,
+  );
+  if (
+    state[name].recipient.nickname === '' &&
+    state.metamask.ensResolutionsByAddress
+  ) {
+    const recipient = {
+      ...state[name].recipient,
+      nickname: getEnsResolutionByAddress(state, checksummedAddress),
+    };
+    return recipient;
+  }
   return state[name].recipient;
```

Comment:
We can simplify this by using `||` which will evaluate an empty string to false, so ``` '' || 'hello' === 'hello'``` so we don't need to check for the presence of a ensResolution first, just checksum the address, and try to get it and let the selector return ```''``` if nothing is found. 
```suggestion
  const checksummedAddress = toChecksumHexAddress(
    state[name].recipient.address,
  );
  return {
    ...state[name].recipient,
    nickname: state[name].recipient.nickname || getEnsResolutionByAddress(state, checksummedAddress),
  }
```

Reply:
Thank you @brad-decker for a suggestion, this is much better than the `if` statement that I used. :)

Reply:
@brad-decker As you can see, `test-unit` are failing. The reason why they failing is that because on the `send.test.js` in `/ui/pages/send` we don't have `ensResolutionsByAddress` added in `baseStore.metamask`.
This is the error that is shown in Circle-CI: `TypeError: Cannot read property '' of undefined`.
When I added that property and run unit tests locally I run into infinity loop and `send.test.js` never finished. 
So I think that we need to use `state.metamask.ensResolutionsByAddress` in the `getRecipient` selector in the send duck to check if the ens resolusion is present. If ens resolusion is present then we return an ens resolution as the nickname. When I added that statement all tests are passed and everything works fine.[Here](https://github.com/MetaMask/metamask-extension/pull/14613/commits/3d3850488b4251f1044fc290a128115e5960302f) is the changes that I made to fix `test-unit` and pipeline to be green.

Reply:
You probably need to update the store object we are using in the failing test to have the ensResolutionByAddress object, which will be present in state in production. @VSaric